Add api/metadata endpoints and support Metadata search (#1284)
web-platform-tests/wpt.fyi: api/README.md

```

</details>

## Metadata results

### /api/metadata

This endpoint accepts POST and GET requests.

- GET request returns Metadata Link Information by product, and requires product parameters;

- POST request searches Metadata Link by link url, and requires product parameters and payload.

> __Parameters__

Reviewer: Clarify these are *URL* params.

__`product`__ : browser[version[os[version]]]. e.g. `chrome-63.0-linux`

> #### Link Query

Reviewer: So IIUC this goes to the body as a JSON, right? Make that clear.

> #### JSON Payload

Reviewer 2: This (`link`) is a query parameter, not part of the JSON payload?

Author: Nope, link is a part of payload. I think it could be either in the query parameter or a part of the JSON payload, but I did so because that is what searchcache does.

Reviewer 2: searchcache has a complicated JSON blob; this example doesn't imply a JSON blob. Is `link` a field on a JSON object?

>   link: [pattern]

Reviewer: For consistency of style:
```suggestion
__`link`__: [pattern]
```

Also remove the spaces at the beginning of next line.

  Where `[pattern]` is a susbtring of the url field of a Metadata Link Node.

#### Get Examples

- /api/metadata?product=chrome&product=safari

<details><summary><b>Example JSON</b></summary>

```json
[
   {
      "test":"/IndexedDB/bindings-inject-key.html",
      "urls":[
         "bugs.chromium.org/p/chromium/issues/detail?id=934844",
         ""
      ]
   },
   {
      "test":"/html/browsers/history/the-history-interface/007.html",
      "urls":[
         "bugs.chromium.org/p/chromium/issues/detail?id=592874",
         ""
      ]
   }
]
```
</details>

#### Post Examples
- POST /api/metadata?product=chrome\&product=firefox \

>     run_ids:="[1, 2, 3]" query:='{"exists":[{"link":"bugs.chromium.org"}]}'

Author: update this

```

</details>

## Metadata results

### /api/metadata

This endpoint accepts POST and GET requests.

- GET request returns Metadata Link Information by product, and requires product parameters;

- POST request searches Metadata Link by link url, and requires product parameters and payload.

__URL Parameters__

__`product`__ : browser[version[os[version]]]. e.g. `chrome-63.0-linux`

#### JSON Request Payload
```json
[
  {

>     "link": [pattern]

Reviewer 2: ```suggestion
    "link": "[pattern]"
```

  }
]
```

>   Where `[pattern]` is a susbtring of the url field of a Metadata Link Node.

Reviewer 2: ```suggestion
  Where `[pattern]` is any substring of the url field of a wpt-metadata `link` node.
```

#### Get Examples

- /api/metadata?product=chrome&product=safari

<details><summary><b>Example JSON</b></summary>

```json
[
   {
      "test":"/IndexedDB/bindings-inject-key.html",
      "urls":[
         "bugs.chromium.org/p/chromium/issues/detail?id=934844",

>          ""

Reviewer 2: what's with the empty strings? (here and L801)

Author: I added a url link for safari. It was empty because this test didn't have a corresponding bug link in safari. But I guess if it is just for demonstration, the link doesn't have to be real?
